pkg/apis/core/types.go

```diff
@@ -4078,10 +4078,19 @@ type ServicePort struct {
 	Protocol Protocol
 
 	// The application protocol for this port.
+	// This is used as a hint for implementations to offer richer behavior for protocols that they understand.
 	// This field follows standard Kubernetes label syntax.
-	// Un-prefixed names are reserved for IANA standard service names (as per
+	// Valid values are either:
+	//
+	// * Un-prefixed protocol names - reserved for IANA standard service names (as per
 	// RFC-6335 and https://www.iana.org/assignments/service-names).
-	// Non-standard protocols should use prefixed names such as
+	//
+	// * Kubernetes-defined prefixed names:
+	//   * 'kubernetes.io/h2c' - http2 over cleartext as described in https://www.rfc-editor.org/rfc/rfc7540
+	//   * 'kubernetes.io/grpc' - grpc traffic - see https://github.com/grpc/grpc/blob/v1.51.1/doc/PROTOCOL-HTTP2.md
```

Comment:
gRPC over TLS or over plaintext?

Reply:
It may be sufficient to state "gRPC over HTTP/2" to align with the corresponding spec: https://github.com/grpc/grpc/blob/master/doc/PROTOCOL-HTTP2.md. That spec does not seem to explicitly require TLS, just that "The HTTP2 specification mandates the use of TLS 1.2 or higher when TLS is used with HTTP2".

Reply:
or you mean "gRPC over HTTP/2" as not "grpc-web"?

Reply:
That doc implies its over HTTP2-over-TLS, but practically speaking gRPC over h2c is a real thing used in a variety of places, so a small comment would go a long ways I think.